library/x509_crl.c

```diff
@@ -294,7 +294,7 @@ int mbedtls_x509_crl_parse_der( mbedtls_x509_crl *chain,
     /*
      * Copy raw DER-encoded CRL
      */
-    if( ( p = mbedtls_calloc( 1, buflen ) ) == NULL )
+    if( buflen != 0 && ( ( p = mbedtls_calloc( 1, buflen ) ) == NULL ) )
```

Comment:
The check for `buflen == 0` should be made earlier. See https://github.com/ARMmbed/mbedtls/pull/1009/files#r153007520